doc/snmp/snmp-configdb-migration-hld.md

```diff
@@ -0,0 +1,221 @@
+Community String    Community Type
+------------------  ----------------
+Qi                  RO
+Travis              RO
+Bill                RO
+Jack                RW
+public              RO
+Joker               RW
+admin@str-s6000-acs-11:~$ 
+```
+
+show run snmp contact
+```
+admin@str-s6000-acs-11:~$ show run snmp contact
+Contact    Contact Email
+---------  -----------------
+Joe        joe@microsoft.com
+admin@str-s6000-acs-11:~$ 
+```
+
+show run snmp location
+```
+admin@str-s6000-acs-11:~$ show run snmp location 
+Location
+----------
+Redmond
+admin@str-s6000-acs-11:~$ 
+```
+
+show run snmp users
+```
+admin@str-s6000-acs-11:~$ show run snmp users
+User    Type    Auth Type    Auth Password    Encryption Type    Encryption Password
+------  ------  -----------  ---------------  -----------------  ---------------------
+Travis  Priv    SHA          TravisAuthPass   AES                TravisEncryptPass
+admin@str-s6000-acs-11:~$
+```
+
+
+# Config Commands
+```
+admin@str-s6000-acs-11:~$ sudo config snmp -h
+Usage: config snmp [OPTIONS] COMMAND [ARGS]...
+
+  SNMP configuration tasks
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  community
+  contact
+  location
+  user
+admin@str-s6000-acs-11:~$ 
+```
+
+sudo config snmp community 
+```
+admin@str-s6000-acs-11:~$ sudo config snmp community -h
+Usage: config snmp community [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  add  Add snmp community
+  del  Add snmp community
+admin@str-s6000-acs-11:~$ 
+```
+
+sudo config snmp contact  
+```
+admin@str-s6000-acs-11:~$ sudo config snmp contact -h
+Usage: config snmp contact [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  modify  Modify snmp contact
+admin@str-s6000-acs-11:~$ 
+```
+
+sudo config snmp location 
+```
+admin@str-s6000-acs-11:~$ sudo config snmp location -h
+Usage: config snmp location [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  modify  Modify snmp location
+admin@str-s6000-acs-11:~$ 
+```
+
+sudo config snmp user 
+```
+admin@str-s6000-acs-11:~$ sudo config snmp user -h
+Usage: config snmp user [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  add  Add snmp user
+  del  Add snmp user
+admin@str-s6000-acs-11:~$ 
+```
+
+# Migration Plan
+In order to move from using the snmp.yml file to using the Redis ConfigDB here are the things the need to be done to move this in a way that is backward compatible. 
+1. Create a python conversion script to parse the data in snmp.yml file and store it in ConfigDB using the above schema.
+2. Update Dockerfile.j2 in docker-snmp container to add a line to copy over new python conversion script to "/usr/bin/"
+3. Update the snmpd.conf.j2 jinja template to pull SNMP information from only the Redis ConfigDB
```

Comment:
Initially, The SNMP information will be pulled from both configDB and snmp.yml, as mentioned in the PR: https://github.com/Azure/sonic-buildimage/pull/6134. So step 3 should be to use both till complete migration is done, is that right?

Reply:
Per my conversation with Suvarna with this setup we will just be able to pull the snmp.yml info into configdb with the conversion script when the docker container starts.  This way we will always get the info from the snmp.yml file but allow us to use the configdb.   This way the jinja template only has to support the configdb since the snmp.yml file info was already imported we will get both.